src/library/dao/playlistdao.cpp

```diff
@@ -150,6 +150,26 @@ QList<TrackId> PlaylistDAO::getTrackIds(const int playlistId) const {
     return trackIds;
 }
 
+QList<TrackId> PlaylistDAO::getTrackIdsInPlaylistOrder(const int playlistId) const {
```

Comment:
This method is a copy of `getTrackIds()`, but with explicit ordering of the returned track ids.  I couldn't see any equivalent method that already does this, and didn't want to break existing behaviour.

I've kept the `DISTINCT` in here, but I'm not sure why it was in the original `getTrackIds()` method.  Happy to be enlightened by someone in the know!